barretenberg/cpp/src/barretenberg/bb/main.cpp

```diff
@@ -78,19 +78,22 @@ bool proveAndVerify(const std::string& bytecodePath, const std::string& witnessP
 void prove(const std::string& bytecodePath,
            const std::string& witnessPath,
            bool recursive,
-           const std::string& outputPath)
+           const std::string& outputProofPath)
 {
     auto acir_composer = new acir_proofs::AcirComposer(MAX_CIRCUIT_SIZE, verbose);
     auto constraint_system = get_constraint_system(bytecodePath);
     auto witness = get_witness(witnessPath);
-    auto proof = acir_composer->create_proof(srs::get_crs_factory(), constraint_system, witness, recursive);
+    auto [proof_without_public_inputs, public_inputs] =
+        acir_composer->create_proof_public_splitted(srs::get_crs_factory(), constraint_system, witness, recursive);
 
-    if (outputPath == "-") {
-        writeRawBytesToStdout(proof);
-        vinfo("proof written to stdout");
+    if (outputProofPath == "-") {
+        writeRawBytesToStdout(proof_without_public_inputs);
+        writeRawBytesToStdout(public_inputs);
+        vinfo("proof and public inputs written to stdout");
     } else {
-        write_file(outputPath, proof);
-        vinfo("proof written to: ", outputPath);
+        write_file(outputProofPath, proof_without_public_inputs);
+        write_file(outputProofPath + "-public_inputs", public_inputs);
```

Comment:
This is _one_ way to do it, we could also pass in a path for public inputs, but don't think you'd want the proof in /kev/foo and then public inputs in /kev/bar